HW04.Rmd

```diff
@@ -69,51 +138,143 @@ Assume that:
 * Fall = October-December
 
 ```{r season}
-
+#Adds season column to gun_deaths_by_month using cut() function
+gun_deaths_by_month$season <- cut(gun_deaths_by_month$month, c(0,3.5,6.5,9.5,13), 
+                                  c("Winter", "Spring", "Summer", "Fall"))
+
+#Adds deaths per season and year
+gun_deaths_by_month <- gun_deaths_by_month %>% 
+    group_by(season, year) %>%
+    mutate(deaths_season = sum(deaths_month)) %>%
+    mutate(avg_deaths_season = mean(deaths_season)) %>% 
+    ungroup()
+ 
+#Produce table to look at values   
+gun_deaths_by_month %>%
+    select(year, season, deaths_season) %>%
+    distinct() %>%
+    arrange(year)
+
+#Generates barplot of deaths
+gun_deaths_by_month %>%
+    arrange(desc(year)) %>%
+    distinct() %>% 
+    ggplot(mapping = aes(x = season, y = deaths_month, fill = as.character(year))) +
+    geom_bar(stat = "identity", alpha = 0.7) + 
+    scale_y_continuous(limits = c(0,29000)) +
+    scale_fill_brewer(palette = "Set2") +
+    xlab("Season") + ylab("Gun Death Victims") +
+    ggtitle("Season and Year of Gun Deaths in America (2012-2014)") +
+    theme(legend.title = element_blank(), legend.position = c(0.5, 0.93),
+          legend.background = element_blank(), legend.direction = "horizontal", 
+          plot.title = element_text(hjust = 0.5))
 ```
 
 
 # These are more open ended questions from here on out, you could look at making multiple summaries or types of plots. 
 
 ### Are whites who are killed by guns more likely to die because of suicide or homicide? How does this compare to blacks and hispanics?
 
-```{r race}
-
+```{r race whites suicide prevalence}
+#Creates table based on determined gun death intent and race
+gun_deaths_by_race <- gun_deaths %>%
+    select(intent, race) %>%
+    group_by(intent, race) %>%
+    add_count(intent, name = "deaths_intent") %>%
+    ungroup() %>%
+    distinct() %>% #Removes duplicate rows
+    group_by(race) %>% 
+    mutate(pop_deaths_race = sum(deaths_intent)) %>%
+    mutate(per_int_race = deaths_intent/ pop_deaths_race * 100) %>% 
+    ungroup()
+
+gun_deaths_by_race %>%
+    filter(race %in% c("Black", "Hispanic", "White")) %>% 
+    ggplot(mapping = aes(x = "", y = per_int_race, fill = intent, group = intent)) +
+    geom_bar(stat = "identity", color = "white") + facet_grid(cols = vars(race)) + 
+    coord_polar("y") + scale_fill_brewer(palette = "Set1") + theme_void() + 
+    ggtitle("Percent of Gun Deaths by Race and Intent (2012-2014)") +
+    theme(legend.title = element_blank(), legend.direction = "horizontal",
+          legend.position = c(0.5, -0.1), plot.title = element_text(hjust = 0.5))

```

Comment:
For your pie charts, the "Not Available" category is not useful since it is only 1 death out of thousands and might just be an error in the data. Excluding that datum might be make the plot look a bit cleaner